### DISCUSSION TITLE
Getting 403 Forbidden Error on clicking a Submit button

### DISCUSSION THREAD

**Discussion by arunrg19** [cypress-io/cypress#26548] [Questions and Help]
In my application, I'm just updating an user profile and clicking on Save. On Save button click cypress is giving **"(stub-8)error("FAILED TO SAVE DATA. TRY AGAIN", "Forbidden")"**

Manually I'm not getting this error. Only through cypress I'm getting this issue and the details are not being saved.
![Cypress 403 Error](https://user-images.githubusercontent.com/107926490/233262252-abd07967-2879-4b6a-9de8-ae591803df6d.jpg)
![image](https://user-images.githubusercontent.com/107926490/233264980-b75ccd1d-89dc-4ba6-88c8-a815c2fae450.png)

Since the data updation is not saved, the scripts are failing. Please help me with this.
Thank you!

**Comment by portyu9** [comment]
Hi @arunrg19 

The screenshots show this is not a Cypress `.click()` problem. The POST to `/bin/user/saveprofile` is reaching the server and returning `403 Authorization Failed`.

If this is AEM, make sure the Cypress session is authenticated and that the AEM CSRF token is loaded and sent with the POST. AEM rejects authenticated POST requests without a valid CSRF token with `403 Forbidden`.

Verify that `/libs/granite/csrf/token.json` succeeds before clicking Save and that the resulting request contains the `CSRF-Token` header.

Do not use `{ force: true }` or suppress the error—the fix is restoring the authenticated session/CSRF token on the save request.

**Reply by portyu9** [reply]
@arunrg19 Please mark this answer as accepted if it helped, thank you.
